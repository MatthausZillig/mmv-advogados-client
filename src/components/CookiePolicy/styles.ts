import styled, { css, DefaultTheme } from 'styled-components'
import { CookieProps } from '.'

const WrapperModifiers = {
  isClose: (theme: DefaultTheme) => css`
    display: none;
    color: #fff;
  `
}

export const Wrapper = styled.main<CookieProps>`
  ${({ theme, close }) => css`
    position: fixed;
    bottom: 20px;
    right: 20px;
    box-shadow: 0 1px 4px rgba(0, 0, 0, 0.2);
    padding: ${theme.spacings.small};
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    width: 400px;
    height: 200px;
    z-index: ${theme.layers.alwaysOnTop};
    background: ${theme.colors.lightBlue};
    ${!!close && WrapperModifiers.isClose(theme)}
    p {
      margin: 0;
    }
  `}
`
