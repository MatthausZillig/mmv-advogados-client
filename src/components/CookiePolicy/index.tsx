import { useState } from 'react'
import * as S from './styles'
import Button from 'components/Button'

export type CookieProps = {
  close: boolean
}

const CookiePolicy = () => {
  const [isclose, setIsClose] = useState(false)

  const handleClose = () => {
    setIsClose(!isclose)
    console.log('close', isclose)
  }

  return (
    <S.Wrapper close={isclose}>
      <div>
        <p>Este site salva seu histórico de uso.</p>
        <p>
          Ao continuar navegando você concorda com a política de privacidade.
        </p>
      </div>
      <div
        style={{
          display: 'flex',
          justifyContent: 'space-evenly',
          width: '100%'
        }}
      >
        <Button onClick={handleClose}>Aceitar</Button>
        <Button withBorder onClick={handleClose}>
          Fechar
        </Button>
      </div>
    </S.Wrapper>
  )
}

export default CookiePolicy
