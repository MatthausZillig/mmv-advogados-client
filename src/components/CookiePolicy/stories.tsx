import { Story, Meta } from '@storybook/react/types-6-0'
import CookiePolicy from '.'

export default {
  title: 'CookiePolicy',
  component: CookiePolicy
} as Meta

export const Default: Story = () => <CookiePolicy />
