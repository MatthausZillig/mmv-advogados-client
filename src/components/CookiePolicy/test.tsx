import { render, screen } from '@testing-library/react'

import CookiePolicy from '.'

describe('<CookiePolicy />', () => {
  it('should render the heading', () => {
    const { container } = render(<CookiePolicy />)

    expect(screen.getByRole('heading', { name: /CookiePolicy/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
