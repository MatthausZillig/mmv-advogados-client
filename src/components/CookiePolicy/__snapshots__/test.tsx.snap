// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CookiePolicy /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    CookiePolicy
  </h1>
</main>
`;
